sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -347,6 +347,54 @@ RT::PiProgram ProgramManager::createPIProgram(const RTDeviceBinaryImage &Img,
 
   return Res;
 }
+static void applyOptionsFromImage(std::string &CompileOpts,
+                                  std::string &LinkOpts,
+                                  const RTDeviceBinaryImage &Img) {
+  // Build options are overridden if environment variables are present.
+  // Environment variables are not changed during program lifecycle so it
+  // is reasonable to use static here to read them only once.
```

Comment:
The comment was copied from previous implementation as well as static ness. From my perspective it make sense as:
 - staticness allows to avoid extra call;
 - we can change environment variable from program itself (using setenv) but it does not have much sense.